### ISSUE TITLE
[REQ] PHP generator to support "repository" param

### ISSUE THREAD

**Issue by dkarlovi** [OpenAPITools/openapi-generator#7780]
### Is your feature request related to a problem? Please describe.

Currently, the `php` package does all sorts of things;
- generates `autoload.php` and suggests [Manual installation](https://github.com/flexolabs/reddit-client#manual-installation)
- suggests [adding the repository into your `composer.json`](https://github.com/flexolabs/reddit-client#composer)

This is unnecessary if your package is [correctly registered at Packagist](https://packagist.org/packages/flexolabs/reddit-client). 

## Describe the solution you'd like

The modern PHP way would be to suggest just

```
composer require flexolabs/reddit-client
```

And you're done, no autoload, nothing.

Of course, to keep BC, a new param would be introduced to allow users to specify what method of delivery they use, for example

1,  `--repository=composer`, regular Packagist, nothing added and the README is the most clean
2. `--repository=composer+https://repo.packagist.com/myorg/`, using [private Packagist](https://packagist.com/) or [self-hosted Satis](https://getcomposer.org/doc/articles/handling-private-packages.md)
3. `--repository=vcs+https://github.com/flexolabs/reddit-client.git`, same as now, points to a Git repo

Basically, support any sort of [repository Composer provides](https://getcomposer.org/doc/05-repositories.md). If supplied, skip adding `autoload.php` and mentioning it in the README since that time has passed for PHP.

## Additional context

Composer is uniformly used by all the major PHP products and solutions, even Wordpress. Not using it is a relic.

**Comment by Valmoz** [comment]
I too think this is an important feature for PHP generator...
